cpp/include/cuvs/neighbors/cagra.hpp

```diff
@@ -448,6 +445,92 @@ void build_host(raft::resources const& handle,
                 raft::host_matrix_view<const uint8_t, int64_t, raft::row_major> dataset,
                 cuvs::neighbors::cagra::index<uint8_t, uint32_t>& idx);
 
+/**
+ * @defgroup cagra_cpp_index_add_nodes CAGRA graph node addition functions
+ * @{
+ */
+void add_graph_nodes(
```

Comment:
If the intended usage of this feature is going to be for the user to call extend() then I think we should move add_nodes to the internal APIs to simplify things. Is there any other reason to keep this part of the public APIs?

Reply:
Yes, this function is for users who do not want to use host memory a lot. In the new vector addition operation, we need a concatenated dataset of the existing and additional parts. So, the `extend` function allocates memory space for the concatenated dataset inside and copies both parts to it. However, if the dataset is very large, this method is not user-friendly. In this situation, users can add vectors without additional host memory by allocating a larger memory area in anticipation of adding vectors later and then calling this function to update the graph.